usr/lib/common/asn1.c

```diff
@@ -411,8 +411,109 @@ CK_RV ber_decode_OCTET_STRING(CK_BYTE *str,
     return CKR_FUNCTION_FAILED;
 }
 
-//
-//
+/**
+ * Here we assume that the indication about unused bits is NOT
+ * part of the data. It will be added by this function.
+ */
+CK_ULONG ber_encode_BIT_STRING(CK_BBOOL length_only,
```

Comment:
dont we link some library for normal ber endcoding ?

Reply:
We have a dependency on OpenLDAP libs lber and ldap, see configure.ac:
> OPENLDAP_LIBS=
> AC_CHECK_HEADERS([lber.h ldap.h],
>                 [OPENLDAP_LIBS="-llber -lldap"],
>                 [AC_MSG_ERROR([lber.h and ldap.h are missing. Please install
>                               'openldap-devel'.])])
> LIBS="$LIBS $OPENLDAP_LIBS"
> AC_SUBST([OPENLDAP_LIBS])

Reply:
My point was that this lib probably has some interface  for encoding bit strings so we dont have to write code for that ?

Reply:
You mean ber_put_bitstring and ber_flatten that are used at many places in Opencryptoki? In fact they do the same as ber_encode_BIT_STRING, but their usage is more complicated than ber_encode_BIT_STRING. Per Ingos comment I verified that I can replace ber_put_bitstring / ber_flatten by ber_encode_BIT_STRING. After all we can do it either way, but should end up with only one implementation. I think it would be a good idea to look into OpenSSL for this.

Reply:
okay